
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="..">
      
      
        <link rel="next" href="../starting-assay-project/">
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.3, mkdocs-material-9.1.17">
    
    
      
        <title>Research Data Management Guidelines - Brickman Lab</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.26e3688c.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.ecc896b0.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#research-data-management-guidelines" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Brickman Lab" class="md-header__button md-logo" aria-label="Brickman Lab" data-md-component="logo">
      
  <img src="../BrickmanLab.jpeg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Brickman Lab
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Research Data Management Guidelines
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
          
            
            
            
            <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_1">
            
              <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_2" hidden>
                <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
              </label>
            
          
            
            
            
            <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_2">
            
              <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_1" hidden>
                <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
              </label>
            
          
        </form>
      
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/brickmanlab/wiki" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    brickmanlab/wiki
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Brickman Lab" class="md-nav__button md-logo" aria-label="Brickman Lab" data-md-component="logo">
      
  <img src="../BrickmanLab.jpeg" alt="logo">

    </a>
    Brickman Lab
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/brickmanlab/wiki" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    brickmanlab/wiki
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Research Data Management Guidelines
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Research Data Management Guidelines
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1-folder-structure-and-organization" class="md-nav__link">
    1. Folder structure and organization
  </a>
  
    <nav class="md-nav" aria-label="1. Folder structure and organization">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#11-template-engine" class="md-nav__link">
    1.1 Template engine
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#12-assay-folder" class="md-nav__link">
    1.2 Assay folder
  </a>
  
    <nav class="md-nav" aria-label="1.2 Assay folder">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#assay-id-code-names" class="md-nav__link">
    Assay ID code names
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#folder-structure" class="md-nav__link">
    Folder Structure
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#13-project-folder" class="md-nav__link">
    1.3 Project folder
  </a>
  
    <nav class="md-nav" aria-label="1.3 Project folder">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#folder-structure_1" class="md-nav__link">
    Folder structure
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#14-synchronization-with-dangpu-server" class="md-nav__link">
    1.4 Synchronization with DanGPU server
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#15-general-naming-conventions-and-more-info" class="md-nav__link">
    1.5 General naming conventions and more info
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-metadata-and-documentation" class="md-nav__link">
    2. Metadata and documentation
  </a>
  
    <nav class="md-nav" aria-label="2. Metadata and documentation">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#21-assay-metadata-fields" class="md-nav__link">
    2.1 Assay metadata fields
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#22-project-metadata-fields" class="md-nav__link">
    2.2 Project metadata fields
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-data-management-catalogue" class="md-nav__link">
    3. Data management catalogue
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-database-browser" class="md-nav__link">
    4. Database browser
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#5-projects-version-control" class="md-nav__link">
    5. Projects version control
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#6-projects-github-pages" class="md-nav__link">
    6. Projects GitHub pages
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#7-project-archiving-in-zenodo" class="md-nav__link">
    7. Project archiving in Zenodo
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#8-data-upload-to-geo" class="md-nav__link">
    8. Data upload to GEO
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#9-create-a-data-management-plan" class="md-nav__link">
    9. Create a Data Management Plan
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../starting-assay-project/" class="md-nav__link">
        Starting a new assay or project
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../tools/" class="md-nav__link">
        Bioinformatics tools
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
      
      
      
        <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          Dangpu
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Dangpu
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../dangpu/" class="md-nav__link">
        DanGPU
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../dangpu/alphafold2/" class="md-nav__link">
        Alphafold 2
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../dangpu/ku-computer/" class="md-nav__link">
        KU computer setup
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../dangpu/nextflow_tower/" class="md-nav__link">
        Nextflow Tower
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../dangpu/packages/" class="md-nav__link">
        Packages
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1-folder-structure-and-organization" class="md-nav__link">
    1. Folder structure and organization
  </a>
  
    <nav class="md-nav" aria-label="1. Folder structure and organization">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#11-template-engine" class="md-nav__link">
    1.1 Template engine
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#12-assay-folder" class="md-nav__link">
    1.2 Assay folder
  </a>
  
    <nav class="md-nav" aria-label="1.2 Assay folder">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#assay-id-code-names" class="md-nav__link">
    Assay ID code names
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#folder-structure" class="md-nav__link">
    Folder Structure
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#13-project-folder" class="md-nav__link">
    1.3 Project folder
  </a>
  
    <nav class="md-nav" aria-label="1.3 Project folder">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#folder-structure_1" class="md-nav__link">
    Folder structure
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#14-synchronization-with-dangpu-server" class="md-nav__link">
    1.4 Synchronization with DanGPU server
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#15-general-naming-conventions-and-more-info" class="md-nav__link">
    1.5 General naming conventions and more info
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-metadata-and-documentation" class="md-nav__link">
    2. Metadata and documentation
  </a>
  
    <nav class="md-nav" aria-label="2. Metadata and documentation">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#21-assay-metadata-fields" class="md-nav__link">
    2.1 Assay metadata fields
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#22-project-metadata-fields" class="md-nav__link">
    2.2 Project metadata fields
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-data-management-catalogue" class="md-nav__link">
    3. Data management catalogue
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-database-browser" class="md-nav__link">
    4. Database browser
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#5-projects-version-control" class="md-nav__link">
    5. Projects version control
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#6-projects-github-pages" class="md-nav__link">
    6. Projects GitHub pages
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#7-project-archiving-in-zenodo" class="md-nav__link">
    7. Project archiving in Zenodo
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#8-data-upload-to-geo" class="md-nav__link">
    8. Data upload to GEO
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#9-create-a-data-management-plan" class="md-nav__link">
    9. Create a Data Management Plan
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<h1 id="research-data-management-guidelines">Research Data Management Guidelines</h1>
<p>This section provides guidelines for effective research data management within our lab. By adopting these guidelines, we aim to improve data organization and naming conventions, leading to enhanced data governance and research efficiency. The guidelines include the following steps:</p>
<ol>
<li>Adhere to folder structure and naming conventions for <code>Assays</code> and <code>Projects</code> folders.</li>
<li>Add relevant metadata to a <code>metadata.yml</code> in each folder</li>
<li>Create a database from metadata files in <code>Assays</code> and <code>Projects</code> folders.</li>
<li>Visualize database with a <a href="https://panel.holoviz.org/">Panel python app</a>.</li>
<li><code>Projects</code> folders will be version controlled with Github and the <a href="https://github.com/brickmanlab">Brickman organization</a>.</li>
<li><code>Projects</code> reports will be displayed under the <a href="https://gbrickmanlab.github.io">Brickman organization GitHub Pages</a>.</li>
<li><code>Projects</code> will be syncronized and archived in <a href="https://zenodo.org/">Zenodo</a>, which will give a DOI that can be used in a publication.</li>
<li>NGS <code>Assays</code> folder will be uploaded to GEO, with the information provided in the metadata file.</li>
<li>Create a Data Management Plan template that it is prefilled with repetitive information using <a href="https://dmponline.deic.dk/">DMPonline</a></li>
</ol>
<h2 id="1-folder-structure-and-organization">1. Folder structure and organization</h2>
<p>To ensure efficient data management, it is important to establish a consistent approach to organizing research data. We consider the following practices:</p>
<ul>
<li>Folder structure: we aim to a logical and intuitive folder structure that reflects the organization of research projects and experimental data. We use descriptive folder names to make it easy to locate and access specific data files.</li>
<li>Subfolders: Use subfolders to further categorize data based on their contents, such as code notebooks, results, reports, etc. This helps to keep data organized and facilitates quick retrieval.</li>
<li>File naming conventions: implement a standardized file naming convention to ensure consistency and clarity. Use descriptive names that include relevant information, such as type of plots, results tables, etc.</li>
</ul>
<h3 id="11-template-engine">1.1 Template engine</h3>
<p>We are currently using a <a href="https://github.com/cookiecutter/cookiecutter">cookiecutter</a>
template to generate a <a href="https://github.com/brickmanlab/ngs-template">folder structure</a>.
Use <a href="https://github.com/cruft/cruft">cruft</a> when generating assay and project folders to
allow us to validate and sync old templates with the latest version.</p>
<h3 id="12-assay-folder">1.2 Assay folder</h3>
<p>For each NGS experiment there should be an <code>Assay</code> folder that will contain all experimental datasets (raw files and pipeline processed files).
Inside <code>Assay</code> there will be subfolders named after a unique NGS ID and the date it was created:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>&lt;Assay-ID&gt;_YYYYMMDD
</span></code></pre></div>
<h4 id="assay-id-code-names">Assay ID code names</h4>
<ul>
<li><code>CHIP</code>: ChIP-seq</li>
<li><code>RNA</code>: RNA-seq</li>
<li><code>ATAC</code>: ATAC-seq</li>
<li><code>SCR</code>: scRNA-seq</li>
<li><code>PROT</code>: Mass Spectrometry Assay</li>
<li><code>CAT</code>: Cut&amp;Tag</li>
<li><code>CAR</code>: Cut&amp;Run</li>
<li><code>RIME</code>: Rapid Immunoprecipitation Mass spectrometry of Endogenous proteins</li>
</ul>
<!-- **Note:** we do not know much about proteomics and mass spec data, neither imaging data.  -->

<p>For example <code>CHIP_20230101</code> is a ChIPseq assay made on 1st January 2023.</p>
<h4 id="folder-structure">Folder Structure</h4>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>CHIP_20230424
</span><span id="__span-1-2"><a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a>├──<span class="w"> </span>description.yaml
</span><span id="__span-1-3"><a id="__codelineno-1-3" name="__codelineno-1-3" href="#__codelineno-1-3"></a>├──<span class="w"> </span>metadata.yaml
</span><span id="__span-1-4"><a id="__codelineno-1-4" name="__codelineno-1-4" href="#__codelineno-1-4"></a>├──<span class="w"> </span>pipeline.md
</span><span id="__span-1-5"><a id="__codelineno-1-5" name="__codelineno-1-5" href="#__codelineno-1-5"></a>├──<span class="w"> </span>processed
</span><span id="__span-1-6"><a id="__codelineno-1-6" name="__codelineno-1-6" href="#__codelineno-1-6"></a>└──<span class="w"> </span>raw
</span><span id="__span-1-7"><a id="__codelineno-1-7" name="__codelineno-1-7" href="#__codelineno-1-7"></a><span class="w">   </span>├──<span class="w"> </span>.fastq.gz
</span><span id="__span-1-8"><a id="__codelineno-1-8" name="__codelineno-1-8" href="#__codelineno-1-8"></a><span class="w">   </span>└──<span class="w"> </span>samplesheet.csv
</span></code></pre></div>
<ul>
<li><strong>description.yaml</strong>: short and long descriptions of the assay in yaml format.</li>
<li><strong>metadata.yaml</strong>: metadata file for the assay describing different keys (<a href="#21-assay-metadata-fields">see below</a>).</li>
<li><strong>pipeline.md</strong>: description of the pipeline used to process raw data.</li>
<li><strong>processed</strong>: folder with results of the preprocessing pipeline. Contents depend on the pipeline used.</li>
<li><strong>raw</strong>: folder with the raw data.</li>
<li><em>.fastq.gz</em>:In the case of NGS assays, there should be fastq files.</li>
<li><em>samplesheet.csv</em>: file that contains metadata information for the samples. This file is used to run the nf-core pipelines. Ideally, it will also contain a column with info regarding the experimental variables and batches so it can be used for down stream analysis as well.</li>
</ul>
<h3 id="13-project-folder">1.3 Project folder</h3>
<p>There should be another folder called <code>Projects</code> that will contain project information and data analysis.</p>
<p>A project may use one or more assays to answer a scientific question. This should be, for example, all the data analysis related
to a publication.</p>
<p>The project folder should be named after a unique identifier, such as:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>&lt;Project-ID&gt;_YYYYMMDD
</span></code></pre></div>
<p><code>&lt;Project-ID&gt;</code> should be the surname of the owner of the project folder and the publication year, e.g. <code>JARH_etal_20230101</code>.</p>
<h4 id="folder-structure_1"><strong>Folder structure</strong></h4>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a>&lt;Project-ID&gt;_20230424
</span><span id="__span-3-2"><a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a>├──<span class="w"> </span>data
</span><span id="__span-3-3"><a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a>│<span class="w">  </span>├──<span class="w"> </span>assays
</span><span id="__span-3-4"><a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a>│<span class="w">  </span>├──<span class="w"> </span>external
</span><span id="__span-3-5"><a id="__codelineno-3-5" name="__codelineno-3-5" href="#__codelineno-3-5"></a>│<span class="w">  </span>└──<span class="w"> </span>processed
</span><span id="__span-3-6"><a id="__codelineno-3-6" name="__codelineno-3-6" href="#__codelineno-3-6"></a>├──<span class="w"> </span>documents
</span><span id="__span-3-7"><a id="__codelineno-3-7" name="__codelineno-3-7" href="#__codelineno-3-7"></a>├──<span class="w"> </span>notebooks
</span><span id="__span-3-8"><a id="__codelineno-3-8" name="__codelineno-3-8" href="#__codelineno-3-8"></a>├──<span class="w"> </span>README.md
</span><span id="__span-3-9"><a id="__codelineno-3-9" name="__codelineno-3-9" href="#__codelineno-3-9"></a>├──<span class="w"> </span>reports
</span><span id="__span-3-10"><a id="__codelineno-3-10" name="__codelineno-3-10" href="#__codelineno-3-10"></a>│<span class="w">  </span>└──<span class="w"> </span>figures
</span><span id="__span-3-11"><a id="__codelineno-3-11" name="__codelineno-3-11" href="#__codelineno-3-11"></a>├──<span class="w"> </span>requirements.txt
</span><span id="__span-3-12"><a id="__codelineno-3-12" name="__codelineno-3-12" href="#__codelineno-3-12"></a>├──<span class="w"> </span>results
</span><span id="__span-3-13"><a id="__codelineno-3-13" name="__codelineno-3-13" href="#__codelineno-3-13"></a>└──<span class="w"> </span>scripts
</span></code></pre></div>
<ul>
<li><strong>data</strong>: folder that contains symlinks or shortcuts to where the data is, avoiding copying and modification of original files.</li>
<li><strong>documents</strong>: folder containing word documents, slides or pdfs related to the project, such as explanations of the data or project, papers, etc.</li>
<li><strong>notebooks</strong>: folder containing Jupyter, R markdown or Quarto notebooks with the actual data analysis. Using annotated notebooks is ideal for reproducibility and readability purposes. Notebooks should be labeled numerically in order they were created e.g. <code>00_preprocessing</code></li>
<li><strong>README.md</strong>: detailed description of the project in markdown format.</li>
<li><strong>reports</strong>: notebooks rendered as html/docx/pdf versions, ideal for sharing with colleagues and also as a formal report of the data analysis procedure.</li>
<li><em>figures</em>: figures produced upon rendering notebooks. The figures will be saved under a subfolder named after the notebook that created them. This is for provenance purposes so we know which notebook created which figures.</li>
<li><strong>results</strong>: results from the data analysis, such as tables with differentially expressed genes, enrichment results, etc. These results should be saved under a subfolder named after the notebook that created them. This is for provenance purposes so we know which notebook created which results.</li>
<li><strong>scripts</strong>: folder containing helper scripts needed to run data analysis or reproduce the work of the folder</li>
</ul>
<!-- **Note**: maybe we can make an environment folder or put inside the scripts folder a Dockerfile that should give you an environment where you can reproduce the results of the folder? -->

<!-- **Note2**: shouldn't there be also a project_metadata.yaml here as well? also a project_description.yaml too? -->

<h3 id="14-synchronization-with-dangpu-server">1.4 Synchronization with DanGPU server</h3>
<p>We will have to setup a cron job to perform one-way sync between the <code>/projects</code>
folder and <code>NGS_data</code> folder. All the analysis will be done on danGPU server,
with <strong>no exceptions</strong>!</p>
<p>After project is done and published, it will be moved to <code>NGS_data</code>.</p>
<h3 id="15-general-naming-conventions-and-more-info">1.5 General naming conventions and more info</h3>
<ul>
<li>date format: <code>YYYYMMDD</code></li>
</ul>
<p><a href="https://faircookbook.elixir-europe.org/content/recipes/interoperability/transcriptomics-metadata.html#analysis-metadata">Transcriptomics metadata standards and fields</a></p>
<p>More info on naming conventions for different types of files and analysis is in development.</p>
<h2 id="2-metadata-and-documentation">2. Metadata and documentation</h2>
<p>Accurate documentation and metadata play a crucial role in facilitating data discovery and interpretation. Consider the following guidelines:</p>
<ul>
<li>Metadata capture: Record essential metadata for each dataset, including type of experiment, date, organisms, etc. This information provides context and helps others understand and reuse the data effectively.</li>
<li>Readme files: Create readme files for each project or dataset. These files should provide a brief overview of the project, list the files and their descriptions, and explain any specific instructions or dependencies required for data analysis.</li>
</ul>
<h3 id="21-assay-metadata-fields">2.1 Assay metadata fields</h3>
<table>
<thead>
<tr>
<th align="left">Metadata field</th>
<th align="left">Definition</th>
<th align="left">Format</th>
<th align="left">ontology</th>
<th align="left">example</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">assay_ID</td>
<td align="left">Identifier for the assay that is at least unique within the project</td>
<td align="left">&lt;code_name>_YYYYMMDD</td>
<td align="left">nan</td>
<td align="left">CHIP_20200101</td>
</tr>
<tr>
<td align="left">assay_type</td>
<td align="left">The type of experiment performed, eg ATAC-seq or seqFISH</td>
<td align="left">nan</td>
<td align="left">ontology field- e.g. EFO or OBI</td>
<td align="left">ChIPseq</td>
</tr>
<tr>
<td align="left">assay_subtype</td>
<td align="left">More specific type or assay like bulk nascent RNAseq or single cell ATACseq</td>
<td align="left">nan</td>
<td align="left">ontology field- e.g. EFO or OBI</td>
<td align="left">bulk ChIPseq</td>
</tr>
<tr>
<td align="left">owner</td>
<td align="left">Owner of the assay (who made the experiment?).</td>
<td align="left">&lt;First Name> &lt;Last Name></td>
<td align="left">nan</td>
<td align="left">Jose Romero</td>
</tr>
<tr>
<td align="left">platform</td>
<td align="left">The type of instrument used to perform the assay, eg Illumina HiSeq 4000 or Fluidigm C1 microfluidics platform</td>
<td align="left">nan</td>
<td align="left">ontology field- e.g. EFO or OBI</td>
<td align="left">Illumina</td>
</tr>
<tr>
<td align="left">extraction_method</td>
<td align="left">Technique used to extract the nucleic acid from the cell</td>
<td align="left">nan</td>
<td align="left">ontology field- e.g. EFO or OBI</td>
<td align="left">nan</td>
</tr>
<tr>
<td align="left">library_method</td>
<td align="left">Technique used to amplify a cDNA library</td>
<td align="left">nan</td>
<td align="left">ontology field- e.g. EFO or OBI</td>
<td align="left">nan</td>
</tr>
<tr>
<td align="left">external accessions</td>
<td align="left">Accession numbers from external resources to which assay or protocol information was submitted</td>
<td align="left">nan</td>
<td align="left">eg protocols.io, AE, GEO accession number, etc</td>
<td align="left">GSEXXXXX</td>
</tr>
<tr>
<td align="left">codename</td>
<td align="left">Internal code name for easy identification (like a keyword)</td>
<td align="left">nan</td>
<td align="left">nan</td>
<td align="left">Oct4_ChIP</td>
</tr>
<tr>
<td align="left">date</td>
<td align="left">Date of assay creation</td>
<td align="left">YYYYMMDD</td>
<td align="left">nan</td>
<td align="left">20200101</td>
</tr>
<tr>
<td align="left">nsamples</td>
<td align="left">Number of samples analyzed in this assay</td>
<td align="left">&lt;integer></td>
<td align="left">nan</td>
<td align="left">9</td>
</tr>
<tr>
<td align="left">is_paired</td>
<td align="left">Paired fastq files or not</td>
<td align="left">&lt;single OR paired></td>
<td align="left">nan</td>
<td align="left">single</td>
</tr>
<tr>
<td align="left">pipeline</td>
<td align="left">Pipeline used to process data</td>
<td align="left">nan</td>
<td align="left">nan</td>
<td align="left">nf-core/chipseq</td>
</tr>
<tr>
<td align="left">strandedness</td>
<td align="left">The strandedness of the cDNA library</td>
<td align="left">&lt;+ OR - OR *></td>
<td align="left">nan</td>
<td align="left">*</td>
</tr>
<tr>
<td align="left">processed_by</td>
<td align="left">Who processed the data</td>
<td align="left">&lt;First Name> &lt;Last Name></td>
<td align="left">nan</td>
<td align="left">Sarah Lundregan</td>
</tr>
<tr>
<td align="left">organism</td>
<td align="left">Organism origin</td>
<td align="left">&lt;Genus species></td>
<td align="left">nan</td>
<td align="left">Mus musculus</td>
</tr>
<tr>
<td align="left">origin</td>
<td align="left">If it is internal or external data</td>
<td align="left">&lt;internal OR external></td>
<td align="left">nan</td>
<td align="left">nan</td>
</tr>
<tr>
<td align="left">path</td>
<td align="left">Path to files</td>
<td align="left">&lt;/path/to/file></td>
<td align="left">nan</td>
<td align="left">nan</td>
</tr>
<tr>
<td align="left">short_desc</td>
<td align="left">Short description of the assay</td>
<td align="left">plain text</td>
<td align="left">nan</td>
<td align="left">Oct4 ChIP after pERK activation</td>
</tr>
</tbody>
</table>
<h3 id="22-project-metadata-fields">2.2 Project metadata fields</h3>
<p>In development.</p>
<h2 id="3-data-management-catalogue">3. Data management catalogue</h2>
<p>@SLundregan is in the process of building a prototype for <code>Assay</code>, using the metadata contained in all <code>description.yml</code> and <code>metadata.yml</code> files in the assay folder.
This will be in the form of an SQLite database that that is easily updatable by running a helper script.</p>
<h2 id="4-database-browser">4. Database browser</h2>
<p>@SLundregan is also working on a browsable database using <a href="https://panel.holoviz.org/">Panel python app</a>.
The app will display the latest version of the SQLite database. Clicking on an item from the database
will open a tab containing all available metadata for the assay.</p>
<p>Also, it would be nice if you can create an <code>Assay</code> folder directly from there,
making it easy to fill up the info for the metadata and GEO submission (<a href="#8-data-upload-to-geo">see below</a>)</p>
<p>In the future, you could ideally visualize an analysed single cell RNAseq dataset by opening <a href="https://cirrocumulus.readthedocs.io/en/latest/">Cirrocumulus</a> session.</p>
<h2 id="5-projects-version-control">5. <code>Projects</code> version control</h2>
<p>All projects should be version controlled using GitHub under the <a href="https://github.com/brickmanlab/">Brickman organization</a>. After creating a cookiecutter template, initiate a git repository on the folder. The Git repository can stay private until it is ready for publication.</p>
<h2 id="6-projects-github-pages">6. <code>Projects</code> GitHub pages</h2>
<p>Using GitHub pages, it is possible to display your data analyses (or anything related to the project) inside the <code>Projects</code> folder so that they are open to the public in a html format.
This is great for transparency and reproducibility purposes. This can be done after the paper has been made public (it is not possible to do with a private repository without paying).</p>
<p><strong>Info on how this is done should be put here</strong></p>
<h2 id="7-project-archiving-in-zenodo">7. <code>Project</code> archiving in Zenodo</h2>
<p>Before submitting, link the repository to <a href="https://zenodo.org/">Zenodo</a> and then create a Git release. This release will be caught by Zenodo and will give you a DOI that you can submit along the manuscript.</p>
<h2 id="8-data-upload-to-geo">8. Data upload to GEO</h2>
<p>The raw data from NGS experiments will be uploaded to the <a href="https://www.ncbi.nlm.nih.gov/geo/">Gene Expression Omnibus (GEO)</a>.
Whenever a new Assay folder is created, the data owner must fill up the required documentation and information needed to make the GEO submission as smooth as possible.</p>
<!-- **Note**: Probably we should start using Annotare  https://www.ebi.ac.uk/fg/annotare/login/ -->

<h2 id="9-create-a-data-management-plan">9. Create a Data Management Plan</h2>
<p>We are currently working on a DMP template that it is prefilled with repetitive information using <a href="https://dmponline.deic.dk/">DMPonline</a> and the KU guidelines.
This template will contain all the necessary information regarding common practices that we will use, the repositories we use for NGS, etc etc.</p>
<p><strong>more info on what is a DMP and how to use it</strong></p>

  <hr>
<div class="md-source-file">
  <small>
    
      Last update:
      <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-date">June 8, 2023</span>
      
        <br>
        Created:
        <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-date">May 18, 2023</span>
      
    
  </small>
</div>





                
              </article>
            </div>
          
          
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "..", "features": [], "search": "../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.b425cdc4.min.js"></script>
      
    
  </body>
</html>